talisker/testing.py

```diff
@@ -140,12 +151,58 @@ class ServerProcessError(Exception):
 
 class ServerProcess(object):
     """Context mananger to run a server subprocess """
-    def __init__(self, cmd):
+    def __init__(self, cmd, env=None):
         self.cmd = cmd
+        self.env = env
         self.output = []
         self.ps = None
         self._log = None
 
+    def start(self):
+        self.output_file = temp_file()
+        self.reader = open(self.output_file.name, 'r')
```

Comment:
Is this guaranteed to be text and thus not to raise exceptions on Python 3 (answer may well be yes, but just checking)?

Reply:
Yeah, the implicit assumption here is that we're testing against textual log output, as it's a helper for testing talisker programs. So I think the answer is yes :)